backend/services/core/src/datasources/Access.ts

```diff
@@ -61,15 +63,15 @@ export default class AccessAPI extends dbUtils.KnexDataSource {
 
   createDoor(ctx: context.UserContext, input: gql.CreateDoor): Promise<gql.Maybe<gql.Door>> {
     return this.withAccess('core:access:door:create', ctx, async () => {
-      const door = await this.knex<sql.Door>('doors').insert(input).returning('*').first();
+      const door = (await this.knex<sql.Door>('doors').insert(input).returning('*'))[0];
       return door;
     });
   }
 
   removeDoor(ctx: context.UserContext, name: string): Promise<gql.Maybe<gql.Door>> {
     return this.withAccess('core:access:door:delete', ctx, async () => {
-      const door = await this.knex<sql.Door>('doors').where({ name }).delete().returning('*')
-        .first();
+      await this.knex('door_access_policies').where({ door_name: name }).del();
+      const door = (await this.knex<sql.Door>('doors').where({ name }).delete().returning('*'))[0];
```

Comment:
you can do const [door] = (await ....) instead of using [0] at the end

Reply:
@emilwihlander